docs/pair_hmm.tex

```diff
@@ -0,0 +1,101 @@
+\begin{itemize}
+\item $(i, j, M/D/I) \rightarrow (i+1, j+1, M)$: match of read position $i+1$ with haplotype position $j + 1$
+\item $(i, j, M/D) \rightarrow (i, j+1, D)$: deletion after read position $i$ -- haplotype position $j+1$ is deleted
+\item $(i, j, M/I) \rightarrow (i + 1, j, I)$: insertion at read position $i + 1$
+\end{itemize}
+Note that the state label $s$ seems redundant because it can be reconstructed from the sequence of $i$ an $j$.  While this is true, our model treats indel starts differently from indel continuations, and by including the state label we can distinguish these conveniently\footnote{That is, we can tell which type of transition it is by looking back one unit instead of two.  Thus we have a first-order Markov model instead of a second-order Markov model.}.
+
+The read-alignment likelihood has two components.  First is the probability of the sequence of match, insertion, and deletion states, which is
+\begin{equation}
+P(\mc{A}) = \prod_k T_{s_k, s_{k+1}},
+\end{equation}
+where $T$ is a constant matrix of state transition probabilities\footnote{The gap continuation probability corresponds to a phred-scaled quality that is set by the \code{gcpHMM} argument, which is 10 by default, implying that $T_{DD} = T_{II} = 0.1$ and $T_{DM} = T{IM} = 0.9$.  The other transitions are hard-coded based on a phred-scaled indel start quality of 45.  That is, $T_{MD} = T{MI} = 10^{-4.5}$ and $T_{MM} = 1 - 2 \times 10^{-4.5}$.  This fact is somewhat obscured in the code, which gives a position index to the transition matrices.  Nonetheless, the array representing transitions is constant over this index.}.  Next is the emission probability of the read bases given the haplotype bases they align to and the base qualities:
+\begin{equation}
+P(\mc{R} | \mc{A}, \mc{H}) = \prod_k P(r_{i_k} | h_{j_k}, q_{i_k})^{{\rm I}[s_k = M]},
+\end{equation}
+where $r_m$ and $h_n$ are the $m$th read bases and $n$th haplotype base and $q_m$ is the quality of the read's $m$th base.  Note that this only includes alignments in the match state.  The per-base emission is given directly from the definition of base quality:
+\begin{equation}
+P(b_2 | b_1, q) = \left\{ \begin{array}{cc} \epsilon(q)/3 & (b_1 \ne b_2) \\ 1 - \epsilon(q) & (b_1 = b_2)  \end{array} \right.
```

Comment:
What is epsilon? I don't think it's the same epsilon in the HMM diagram.

Reply:
It's the error rate associated with the base quality, as I added just now.

Reply:
Why is`epsilon(q)/3`? Is dividing by 3 empirical?

Reply:
@ronlevine the error probability is the probability for any of 3 possible substitution errors, so we divide by 3 to get the likelihood of any particular one.